hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/table/functional/TestHoodieSparkMergeOnReadTableRollback.java

```diff
@@ -1029,4 +1179,37 @@ private HoodieWriteConfig getWriteConfig(boolean autoCommit, boolean rollbackUsi
             .build());
     return cfgBuilder.build();
   }
+
+  /**
+   * Scenario: data table is updated, no changes to MDT
+   */
+  @Test
+  public void testRollbackWithFailurePreMDT() throws Exception {
```

Comment:
Are these functional tests necessary, i.e., what additional coverage do these tests provide?

Reply:
Based on my understanding these tests are simulating failure in different stages of commit like bef. I don't think these cases are covered in the other functional tests.
The failures are simulated like MDT commit has not yet completed, delta commit completed in MDT but markers not yet cleared, delta commit completed in MDT but markers not yet cleared and rollback fails before updating MDT etc.

Reply:
Got it.  It would be good to generate the table once and manipulate it for different scenarios to save time in generating the test tables.  We can improve these separately,